File: src/tests/jwtAuth.test.js
const request = require('supertest');
const app = require('../app');
const jwt = require('jsonwebtoken');

// Mock user data
const user = { id: 1, username: 'testuser' };

// Mock JWT secret
const JWT_SECRET = 'your_jwt_secret';

describe('JWT Authentication', () => {
  let refreshToken;

  beforeAll(() => {
    // Generate a refresh token
    refreshToken = jwt.sign(user, JWT_SECRET, { expiresIn: '7d' });
  });

  test('should authenticate user and return access token', async () => {
    const response = await request(app)
      .post('/api/auth/login')
      .send({ username: user.username, password: 'password' });

    expect(response.status).toBe(200);
    expect(response.body).toHaveProperty('accessToken');
    expect(response.body).toHaveProperty('refreshToken');
  });

  test('should refresh access token using refresh token', async () => {
    const response = await request(app)
      .post('/api/auth/refresh')
      .send({ refreshToken });

    expect(response.status).toBe(200);
    expect(response.body).toHaveProperty('accessToken');
  });

  test('should fail to refresh access token with invalid refresh token', async () => {
    const response = await request(app)
      .post('/api/auth/refresh')
      .send({ refreshToken: 'invalid_token' });

    expect(response.status).toBe(403);
    expect(response.body).toHaveProperty('error');
  });
});

File: src/tests/jwt_auth_test.py
import unittest
from app import app
import jwt

class JWTAuthTestCase(unittest.TestCase):
    def setUp(self):
        self.app = app.test_client()
        self.app.testing = True
        self.jwt_secret = 'your_jwt_secret'
        self.user = {'id': 1, 'username': 'testuser'}

    def test_login(self):
        response = self.app.post('/api/auth/login', json={'username': self.user['username'], 'password': 'password'})
        self.assertEqual(response.status_code, 200)
        self.assertIn('accessToken', response.json)
        self.assertIn('refreshToken', response.json)

    def test_refresh_token(self):
        refresh_token = jwt.encode(self.user, self.jwt_secret, algorithm='HS256')
        response = self.app.post('/api/auth/refresh', json={'refreshToken': refresh_token})
        self.assertEqual(response.status_code, 200)
        self.assertIn('accessToken', response.json)

    def test_refresh_token_invalid(self):
        response = self.app.post('/api/auth/refresh', json={'refreshToken': 'invalid_token'})
        self.assertEqual(response.status_code, 403)
        self.assertIn('error', response.json)

if __name__ == '__main__':
    unittest.main()